File: HtmlStripper.cs
// HtmlStripper.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.RegularExpressions;

class HtmlStripper
{
    public bool StripComments { get; set; } = true;
    public bool ConvertBr { get; set; } = true;
    public bool DecodeEntities { get; set; } = true;
    public bool CollapseSpaces { get; set; } = true;

    public string Strip(string htmlText)
    {
        string text = htmlText;
        // Remove script and style blocks
        text = Regex.Replace(text, @"<script\b[^>]*>.*?</script>", "", RegexOptions.Singleline | RegexOptions.IgnoreCase);
        text = Regex.Replace(text, @"<style\b[^>]*>.*?</style>", "", RegexOptions.Singleline | RegexOptions.IgnoreCase);
        if (StripComments)
            text = Regex.Replace(text, @"<!--.*?-->", "", RegexOptions.Singleline);
        if (ConvertBr)
            text = Regex.Replace(text, @"<br\s*/?>", "\n", RegexOptions.IgnoreCase);
        // Remove tags
        text = Regex.Replace(text, @"<[^>]+>", "");
        if (DecodeEntities)
            text = DecodeHTMLEntities(text);
        if (CollapseSpaces)
        {
            text = Regex.Replace(text, @"[ \t]+", " ");
            text = Regex.Replace(text, @"\n\s*\n", "\n\n");
            text = text.Trim();
        }
        return text;
    }

    private string DecodeHTMLEntities(string s)
    {
        var entities = new Dictionary<string, string>
        {
            {"&nbsp;", " "}, {"&amp;", "&"}, {"&lt;", "<"}, {"&gt;", ">"},
            {"&quot;", "\""}, {"&#39;", "'"}, {"&apos;", "'"}, {"&copy;", "©"},
            {"&reg;", "®"}, {"&trade;", "™"}, {"&euro;", "€"}, {"&pound;", "£"},
            {"&yen;", "¥"}, {"&cent;", "¢"}, {"&sect;", "§"}, {"&deg;", "°"}
        };
        foreach (var kv in entities)
            s = s.Replace(kv.Key, kv.Value);
        return s;
    }

    public string StripFile(string filepath)
    {
        string content = File.ReadAllText(filepath);
        return Strip(content);
    }

    static void Main()
    {
        var stripper = new HtmlStripper();
        Console.WriteLine("=== HTML Stripper ===");
        while (true)
        {
            Console.WriteLine("\n1. Strip HTML from input text");
            Console.WriteLine("2. Strip HTML from file");
            Console.WriteLine($"3. Toggle options (comments: {stripper.StripComments}, convert_br: {stripper.ConvertBr}, decode_entities: {stripper.DecodeEntities}, collapse_spaces: {stripper.CollapseSpaces})");
            Console.WriteLine("4. Exit");
            Console.Write("Choose: ");
            string choice = Console.ReadLine()?.Trim() ?? "";
            switch (choice)
            {
                case "1":
                    Console.WriteLine("Enter HTML text (end with empty line):");
                    var lines = new List<string>();
                    while (true)
                    {
                        string line = Console.ReadLine() ?? "";
                        if (line == "") break;
                        lines.Add(line);
                    }
                    string htmlText = string.Join("\n", lines);
                    string result = stripper.Strip(htmlText);
                    Console.WriteLine("\nProcessed result:\n" + result);
                    break;
                case "2":
                    Console.Write("Enter file path: ");
                    string fname = Console.ReadLine()?.Trim() ?? "";
                    try
                    {
                        result = stripper.StripFile(fname);
                        Console.WriteLine("\nProcessed result:\n" + result);
                    }
                    catch (Exception e)
                    {
                        Console.WriteLine("Error: " + e.Message);
                    }
                    break;
                case "3":
                    stripper.StripComments = !stripper.StripComments;
                    stripper.ConvertBr = !stripper.ConvertBr;
                    stripper.DecodeEntities = !stripper.DecodeEntities;
                    stripper.CollapseSpaces = !stripper.CollapseSpaces;
                    Console.WriteLine("Options toggled.");
                    break;
                case "4":
                    Console.WriteLine("Goodbye!");
                    return;
                default:
                    Console.WriteLine("Invalid choice.");
                    break;
            }
        }
    }
}

File: html_stripper.rb
# html_stripper.rb
class HTMLStripper
  attr_accessor :strip_comments, :convert_br, :decode_entities, :collapse_spaces

  def initialize
    @strip_comments = true
    @convert_br = true
    @decode_entities = true
    @collapse_spaces = true
  end

  def strip(html_text)
    text = html_text.dup
    # Remove script and style blocks
    text.gsub!(/<script\b[^>]*>.*?<\/script>/mi, '')
    text.gsub!(/<style\b[^>]*>.*?<\/style>/mi, '')
    text.gsub!(/<!--.*?-->/, '') if @strip_comments
    text.gsub!(/<br\s*\/?>/i, "\n") if @convert_br
    text.gsub!(/<[^>]+>/, '')
    text = decode_entities(text) if @decode_entities
    if @collapse_spaces
      text.gsub!(/[ \t]+/, ' ')
      text.gsub!(/\n\s*\n/, "\n\n")
      text.strip!
    end
    text
  end

  def decode_entities(s)
    entities = {
      '&nbsp;' => ' ', '&amp;' => '&', '&lt;' => '<', '&gt;' => '>',
      '&quot;' => '"', '&#39;' => "'", '&apos;' => "'", '&copy;' => '©',
      '&reg;' => '®', '&trade;' => '™', '&euro;' => '€', '&pound;' => '£',
      '&yen;' => '¥', '&cent;' => '¢', '&sect;' => '§', '&deg;' => '°'
    }
    entities.each { |ent, char| s.gsub!(ent, char) }
    s
  end

  def strip_file(filepath)
    content = File.read(filepath)
    strip(content)
  end
end

def main
  stripper = HTMLStripper.new
  puts "=== HTML Stripper ==="
  loop do
    puts "\n1. Strip HTML from input text"
    puts "2. Strip HTML from file"
    puts "3. Toggle options (comments: #{stripper.strip_comments}, convert_br: #{stripper.convert_br}, decode_entities: #{stripper.decode_entities}, collapse_spaces: #{stripper.collapse_spaces})"
    puts "4. Exit"
    print "Choose: "
    choice = gets.chomp.strip
    case choice
    when '1'
      puts "Enter HTML text (end with empty line):"
      lines = []
      loop do
        line = gets.chomp
        break if line.empty?
        lines << line
      end
      html_text = lines.join("\n")
      result = stripper.strip(html_text)
      puts "\nProcessed result:\n#{result}"
    when '2'
      print "Enter file path: "
      fname = gets.chomp.strip
      begin
        result = stripper.strip_file(fname)
        puts "\nProcessed result:\n#{result}"
      rescue => e
        puts "Error: #{e.message}"
      end
    when '3'
      stripper.strip_comments = !stripper.strip_comments
      stripper.convert_br = !stripper.convert_br
      stripper.decode_entities = !stripper.decode_entities
      stripper.collapse_spaces = !stripper.collapse_spaces
      puts "Options toggled."
    when '4'
      puts "Goodbye!"
      break
    else
      puts "Invalid choice."
    end
  end
end

main if __FILE__ == $0
